src/site-furniture/Header/Header.tsx

```diff
@@ -111,8 +124,30 @@ const Header = forwardRef<HTMLElement, HeaderProps>(
     const [activeSubmenuId, setActiveSubmenuId] = useState<string | null>(null);
     const closeTimeoutRef = React.useRef<NodeJS.Timeout | null>(null);
 
+    const [bannerHeight, setBannerHeight] = useState(bannerRef?.current ? bannerRef.current.clientHeight : 0);
+
+    useEffect(() => {
+      const bannerElement = bannerRef?.current;
+      if (!bannerElement) return;
+
+      setBannerHeight(bannerElement.clientHeight);
+      const resizeObserver = new window.ResizeObserver(() => {
+        setBannerHeight(bannerElement.clientHeight);
+      });
```

Comment:
Do we need state? Seems like we could instead set the var directly like

```suggestion
      const resizeObserver = new window.ResizeObserver(() => {
        document.body.style.setProperty("--notification-height", bannerElement.clientHeight)
        // or even just query for the banner's height directly instead of using a ref:
        // document.querySelector(`.${getCommonProps(undefined, 'NotificationBanner').className}`.clientHeight);
      });
```

Then we could remove a number of ref- and style-related props and stuff. This method is probably less React-y but feels much more direct to me. Also I realize `getCommonProps` may be overkill here!